examples/common/maths.adb

```diff
@@ -206,6 +204,45 @@ package body Maths is
     end Scaling_Matrix;
 
     --  ------------------------------------------------------------------------
+
+    function To_3 (Array_4 : Single_Array) return Single_Array is
+        Array_3  : Single_Array (0 .. Array_4'Length * 3 / 4 - 1);
+        Index_3  : Int;
+        Index_4  : Int;
+    begin
+        Index_3 := 0;
+        Index_4 := 0;
+        while Index_4 < Array_4'Length - 1 loop
```

Comment:
Even if you assume `Array_4` starts at `0`, this condition is wrong since you access `Array_4 (Index_4 + 2)` inside the loop body.